File: src/types/workout.ts
export interface Exercise {
  name: string
  sets?: Array<{ reps?: number; weight?: number; duration_seconds?: number }>
  distance_m?: number
  notes?: string
}

export interface WorkoutTemplate {
  id: string
  name: string
  description: string | null
  category: string | null
  exercises_json: string
  created_at: string
  updated_at: string
  deleted_at: string | null
  sync_status: string
}

export interface WorkoutLog {
  id: string
  template_id: string | null
  title: string
  performed_at: string
  duration_minutes: number | null
  calories: number | null
  exercises_json: string
  notes: string | null
  created_at: string
  updated_at: string
  deleted_at: string | null
  sync_status: string
}

export interface CreateWorkoutTemplatePayload {
  name: string
  description?: string | null
  category?: string | null
  exercises_json: string
}

export interface CreateWorkoutLogPayload {
  template_id?: string | null
  title: string
  performed_at: string
  duration_minutes?: number | null
  calories?: number | null
  exercises_json: string
  notes?: string | null
}

File: src-tauri/src/models/workout.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct WorkoutTemplate {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub category: Option<String>,
    pub exercises_json: String,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: Option<String>,
    pub sync_status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct WorkoutLog {
    pub id: String,
    pub template_id: Option<String>,
    pub title: String,
    pub performed_at: String,
    pub duration_minutes: Option<i64>,
    pub calories: Option<i64>,
    pub exercises_json: String,
    pub notes: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: Option<String>,
    pub sync_status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateWorkoutTemplatePayload {
    pub name: String,
    pub description: Option<String>,
    pub category: Option<String>,
    pub exercises_json: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateWorkoutLogPayload {
    pub template_id: Option<String>,
    pub title: String,
    pub performed_at: String,
    pub duration_minutes: Option<i64>,
    pub calories: Option<i64>,
    pub exercises_json: String,
    pub notes: Option<String>,
}
